content/articles/angular12-material-table/index.md

```diff
@@ -0,0 +1,297 @@
+    // ....
+  ],
+  imports: [
+    // .........
+    AppMaterialModule,
+
+  ],
+  // ...
+})
+export class AppModule { }
+```
+
+We've updated our `app.module.ts` as shown above by importing the `AppMaterialModule` module to expose the Material modules we had imported.
+
+### Using Material table to display data
+In this section, I'll show you how to create the Material table for `students`. The table will consist of basic student details such as `names` and `registrations`. Let's start by creating the details interface.
+
+Run the following commands on your project root:
+
+```bash
+ng g i student
+```
+
+This will create an interface in the `app/student.ts` file, proceed and edit it as follows:
+
+```js
+export interface Student {
+  firstName:string;
+  lastName:string;
+  studentEmail:string;
+  course:string;
+  yearOfStudy: bigint;
+  registrationNumber:string;
+}
+```
+
+Now, proceed and create an `assets/students.json` file and define the student details that we will use in our RESTful APIs to represent in a table.
+
+```json
+[
+  {
+  "firstName": "John",
+  "lastName": "Doe",
+  "studentEmail": "johndoe@example.com",
+  "course": "Bsc Software Engineering",
+  "yearOfStudy": 2
+},
+  {
+    "firstName": "Test2",
+    "lastName": "Test2",
+    "studentEmail": "test@example.com",
+    "course": "Bsc Computer Science",
+    "yearOfStudy": 4
+  }]
+```
+
+Create  `services/apiService.ts` service file and add the following:
+
+First, begin by defining the `baseURL` in the environment to make your code look organized and for reusability purposes.
+
+```typescript
+// ...
+
+export const environment = {
+  // ...
+  baseURL:'assets/',
+};
+// ...
+```
+
+Then update the service as shown below:
+
+```typescript
+// ...
+//import student intreface
+import {Student} from "../student";
+//import this to make http requests
+import {HttpClient} from "@angular/common/http";
+//we've defined our base url here in the env
+import {environment} from "../../environments/environment";
+// ....
+//class apiService
+export class ApiService {
+
+  constructor(private httpClient: HttpClient) { }
+
+  /**
+   * This method returns students details
+   */
+  getStudentsInformation(): Observable<Student[]>{
+    return this.httpClient.get<Student[]>(`${environment.baseURL}student.json`);
+  }
+}
+```
+
+Now that we've got the logic to get data from our API, let's now proceed and add this to the controller, but first, let's update our `app.module.ts` file as follows:  
+
+```typescript
+...
+
+import { HttpClientModule} from "@angular/common/http";
+
+@NgModule({
+  declarations: [
+    ...
+  ],
+  imports: [
+    ...
+    HttpClientModule, //endsure your import this module
+  ],
+  ...
+})
+...
+
+```
+
+Then update the `app.component.ts` contents as shown below:
+
+```typescript
+...
+import {MatTableDataSource} from "@angular/material/table";
+import {Student} from "./student";
+import {ApiService} from "./services/api.service";
+
+@Component({
+  ...
+})
```

Comment:
Please comments these: `....`. Go through all your code blocks and comment them.

Reply:
'These ' means that there are missing code snippets that should be added for code to work??? why should I remove them?